prog/main.go

```diff
@@ -45,27 +50,142 @@ func setLogLevel(levelname string) {
 	log.SetLevel(level)
 }
 
-func usage() {
-	fmt.Fprintf(os.Stderr, "usage: %s (app|probe|version) args...\n", os.Args[0])
-	os.Exit(1)
+type flags struct {
+	probe probeFlags
+	app   appFlags
+}
+
+type probeFlags struct {
+	token           string
+	httpListen      string
+	publishInterval time.Duration
+	spyInterval     time.Duration
+	spyProcs        bool
+	procRoot        string
+	pluginsRoot     string
+	useConntrack    bool
+	insecure        bool
+	logPrefix       string
+	logLevel        string
+
+	dockerEnabled  bool
+	dockerInterval time.Duration
+	dockerBridge   string
+
+	kubernetesEnabled  bool
+	kubernetesAPI      string
+	kubernetesInterval time.Duration
+
+	weaveAddr     string
+	weaveHostname string
+}
+
+type appFlags struct {
+	window    time.Duration
+	listen    string
+	logLevel  string
+	logPrefix string
+	logHTTP   bool
+
+	weaveAddr      string
+	weaveHostname  string
+	containerName  string
+	dockerEndpoint string
+
+	collectorURL     string
+	controlRouterURL string
+	pipeRouterURL    string
+	userIDHeader     string
+
+	awsCreateTables bool
+	consulInf       string
```

Comment:
Maybe these should each live in `prog/probe.go`, and `prog/app.go` respectively? with a method to register them? Or both in `prog/flags.go`?